src/globus_cli/parsing/shared_options/endpoint_create_and_update.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import sys
+from typing import Callable, TypeVar, Union
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+if sys.version_info >= (3, 11):
+    from typing import assert_never
+else:
+    from typing_extensions import assert_never
+
+import click
+
+C = TypeVar("C", bound=Union[Callable, click.Command])
+
+
+def _apply_universal_endpointish_params(
+    f: C, name: str, *, display_name: Literal["argument", "option", "null"] = "null"
+) -> C:
+    if display_name == "argument":
+        f = click.argument("DISPLAY_NAME")(f)
+    elif display_name == "option":
+        f = click.option("--display-name", help=f"Name for the {name}")(f)
+    elif display_name == "null":
+        pass
+    else:
+        assert_never()
```

Comment:
first time seeing assert_never for type checking impossible conditions like this, I like it

Reply:
It's very good for enums, Literals, and the new match-case statements. Expect to see more of it from me in the future. :-)